codegen-renderers/src/main/kotlin/io/vrap/codegen/languages/extensions/ListExtensions.kt

```diff
@@ -0,0 +1,44 @@
+package io.vrap.codegen.languages.extensions
+
+private class TopologySortContext<T>(val predecessors: Map<T, List<T>>) {
+
+    fun hasNoPredecessors(node: T) : Boolean {
+        return predecessors[node]?.size == 0
+    }
+
+    fun remove(node : T) : TopologySortContext<T> {
+        return TopologySortContext(predecessors.mapValues { it.value.minus(node) })
+    }
+}
+
+/**
+ * Sorts the list by topology defined by the given predecessor function.
+ * https://en.wikipedia.org/wiki/Topological_sorting
+ *
+ * @param predecessorsOf function that returns the predecessors of the given argument
+ * @throws IllegalArgumentException thrown when the list contains cycles
+ * @return the topologically sorted list
+ */
+fun <T> List<T>.sortedByTopology(predecessorsOf: (T) -> List<T>) : List<T> {
```

Comment:
I like the Doc here, it makes following the implementation much understandable 🥇